### PR TITLE
Add WindowManagerTools::active_application_zone()

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Add WindowManagerTools::active_application_zone(). (Fixes: #1530)

Also uses it in the miral-shell "floating" window manager.

### CODE REVIEW

File: include/miral/miral/window_manager_tools.h [hunk 1]
```diff
@@ -163,6 +163,10 @@ class WindowManagerTools
     /// Find the active output area
     auto active_output() -> mir::geometry::Rectangle const;
 
+    /// Find the active zone area
+    /// \remark Since MirAL 3.0
+    auto active_zone() -> mir::geometry::Rectangle const;
```
Comment (wmww) [inline include/miral/miral/window_manager_tools.h:168]: Wouldn't it make more sense to return a `miral::Zone`? The policy may want to match the returned zone to an existing application zone it knows about.

Also, can we change the name to `active_application_zone()`? That would clearly indicate that the returned value will be one of the application zones the policy knows about.

**Comment by AlanGriffiths** [comment]
On Mon, 22 Jun 2020 13:44:34 -0700, William Wold wrote:
> @wmww requested changes on this pull request.
>
> When we get to break the API I want to rethink how the 
> `miral::Zone` class works, but until then 

We are breaking ABI.

> we should use it for 
> what it was built for
>
>> @@ -163,6 +163,10 @@ class WindowManagerTools
>      /// Find the active output area
>      auto active_output() -> mir::geometry::Rectangle const;
>  
> +    /// Find the active zone area
> +    /// \remark Since MirAL 3.0
> +    auto active_zone() -> mir::geometry::Rectangle const;
>
> Wouldn't it make more sense to return a `miral::Zone`? The 
> policy may want to match the returned zone to an existing 
> application zone it knows about.
>
> Also, can we change the name to `active_application_zone()`? 
> That would clearly indicate that the returned value will be one 
> of the application zones the policy knows about.
>

-- 
Sent using Dekko from my Ubuntu device

**Comment by wmww** [comment]
In that case I would like to talk about the `Zone` class, as I think it has some issues

**Comment by AlanGriffiths** [comment]
> In that case I would like to talk about the `Zone` class, as I think it has some issues

Here? On the forum? ...?

**Pull request merged**